experiment/dnscheck/dnscheck_test.go

```diff
@@ -0,0 +1,128 @@
+package dnscheck
+
+import (
+	"context"
+	"net/url"
+	"strings"
+	"testing"
+
+	"github.com/apex/log"
+	"github.com/ooni/probe-engine/internal/mockable"
+	"github.com/ooni/probe-engine/model"
+)
+
+func TestExperimentNameAndVersion(t *testing.T) {
+	measurer := NewExperimentMeasurer(Config{Domain: "example.com"})
+
+	if measurer.ExperimentName() != "dnscheck" {
+		t.Error("unexpected experiment name")
+	}
+
+	if measurer.ExperimentVersion() != "0.0.1" {
+		t.Error("unexpected experiment version")
+	}
+}
+
+func TestDNSCheckFailsWithoutInput(t *testing.T) {
+	measurer := NewExperimentMeasurer(Config{Domain: "example.com"})
+
+	err := measurer.Run(
+		context.Background(),
+		newsession(),
+		new(model.Measurement),
+		model.NewPrinterCallbacks(log.Log),
+	)
+
+	if err == nil || !strings.Contains(err.Error(), "this experiment needs input") {
+		t.Fatal("expected no input error")
+	}
+}
+
+func TestDNSCheckFailsInvalidInput(t *testing.T) {
+	measurer := NewExperimentMeasurer(Config{})
+
+	// test with invalid URL
+	err := measurer.Run(
+		context.Background(),
+		newsession(),
+		&model.Measurement{Input: "Not a valid URL \x7f"},
+		model.NewPrinterCallbacks(log.Log),
+	)
+
+	if err == nil || !strings.Contains(err.Error(), "the input URL is invalid") {
```

Comment:
BTW: it just occurred to me that we can simplify this check and the following check if we have two distinct errors rather than assembling errors with the same base error using `fmt.Errorf`. I'd leave to you to decide whether we want to apply this simplification or we're fine with the current code, as it seems to me these two solutions are ~same.